<?xml version="1.0"?>
<launch>
<include file="$(find openni2_launch)/launch/openni2.launch"/>
<include file="$(find terabot_arm_hardware_interface)/launch/move_arm.launch"/>
<include file="$(find arm_calibration)/launch/ar_track_alvar.launch"/>
<node name="arm_calibration" pkg="arm_calibration" type="arm_calibration" output="screen">
<!-- CONFIG PARAMETERS -->
<param name="number_of_points" value="8" /> <!-- Minimum number of samples for the transformation calculation -->
<param name="arm_base_link" value="arm_base_link" /> <!-- arm base frame -->
<param name="camera_link" value="camera_link" /> <!-- camera frame -->
<param name="end_effector" value="end_effector" /> <!-- end effector frame, where the AR marker should be placed -->
<param name="marker_link" value="ar_marker_4" /> <!-- marker id -->
<!-- workspace (arm base_link frame) from which the samples are taken (in meters) -->
<param name="min_x" value="0.4" />
<param name="max_x" value="1.0" />
<param name="min_y" value="-0.7" />
<param name="max_y" value="0.7" />
<param name="min_z" value="0.2" />
<param name="max_z" value="1.0" />
<param name="range" value="0.2" /> <!-- end effector cartesian position workspace from which the samples are taken (in meters) -->
<param name="roll_angle_range" value="0.001" /> <!-- end effector roll orientation range [-value,value] (radians)-->
<param name="roll_angle_offset" value="-1.5" /> <!-- end effector roll orientation offset (radians)-->
<param name="pitch_angle_range" value="1.2" /> <!-- end effector pitch orientation range [-value,value] (radians)-->
<param name="pitch_angle_offset" value="-0.1" /> <!-- end effector roll orientation offset (radians)-->
<param name="yaw_angle_range" value="1.2" /> <!-- end effector yaw orientation range [-value,value] (radians)-->
<param name="yaw_angle_offset" value="-0.1" /> <!-- end effector roll orientation offset (radians)-->
</node>
</launch>
